pkg/pgmodel/migrate.go

```diff
@@ -122,7 +122,7 @@ func CheckSchemaVersion(ctx context.Context, conn *pgx.Conn, versionInfo Version
 		return fmt.Errorf("failed to check schema version: %w", err)
 	}
 	if dbVersion.Compare(expectedVersion) != 0 {
-		return fmt.Errorf("db schema version is incorrect: expected %v, got %v", expectedVersion, dbVersion)
+		return fmt.Errorf("error while comparing schema version. This can cause issues while schema migration. Received schema version %v but expected %v. Please note that no Promscale connectors are running that can prevent from schema migration due to unavailability of schema lock", expectedVersion, dbVersion)
```

Comment:
We should only print this message if we couldn't get the schema lock. That means we need to pass the variable that says whether or not the exclusive schema lock was obtained down to here